libsplinter/src/biome/rest_api/actix/user.rs

```diff
@@ -314,27 +318,54 @@ fn add_delete_user_method(
             }
         };
 
-        Box::new(match user_store.remove_user(&user_id) {
-            Ok(()) => HttpResponse::Ok()
-                .json(json!({ "message": "User deleted sucessfully" }))
-                .into_future(),
-            Err(err) => match err {
-                UserStoreError::NotFoundError(msg) => {
-                    debug!("User not found: {}", msg);
+        match user_store.remove_user(&user_id) {
+            Ok(()) => (),
+            Err(UserStoreError::NotFoundError(msg)) => {
+                debug!("User not found: {}", msg);
+                return Box::new(
                     HttpResponse::NotFound()
                         .json(ErrorResponse::not_found(&format!(
                             "User ID not found: {}",
                             &user_id
                         )))
-                        .into_future()
-                }
-                _ => {
-                    error!("Failed to delete user in database {}", err);
+                        .into_future(),
+                );
+            }
+            Err(err) => {
+                error!("Failed to delete user in database {}", err);
+                return Box::new(
                     HttpResponse::InternalServerError()
                         .json(ErrorResponse::internal_error())
-                        .into_future()
-                }
-            },
-        })
+                        .into_future(),
+                );
+            }
+        };
+
+        match credentials_store.remove_credentials(&user_id) {
+            Ok(()) => Box::new(
+                HttpResponse::Ok()
+                    .json(json!({ "message": "User deleted sucessfully" }))
+                    .into_future(),
+            ),
+            Err(CredentialsStoreError::NotFoundError(msg)) => {
+                debug!("User not found: {}", msg);
+                Box::new(
+                    HttpResponse::NotFound()
+                        .json(ErrorResponse::not_found(&format!(
+                            "User ID not found: {}",
+                            &user_id
+                        )))
+                        .into_future(),
+                )
+            }
```

Comment:
Is it a valid state for a user to exist in the user store but not the credential store?

Reply:
Currently, the credentials feature needs to be enabled for the register and user endpoints to be created, so it shouldn't be a valid state. Are you suggesting this should be an internal error instead?

Reply:
Maybe; if biome should always have a corresponding entry in the credentials store for each user in the users store, and the REST API actions don't let the consumer mess that up, then it would be internal error (something's wrong with the code or the stores).

Reply:
I agree with that. Updated that block to only return internal server errors in error case